x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/slowlog/SecuritySlowLogFieldProvider.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.slowlog;
+
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.SlowLogFieldProvider;
+import org.elasticsearch.xpack.security.Security;
+
+import java.util.Map;
+
+import static org.elasticsearch.index.IndexingSlowLog.INDEX_INDEXING_SLOWLOG_INCLUDE_USER_SETTING;
+import static org.elasticsearch.index.SearchSlowLog.INDEX_SEARCH_SLOWLOG_INCLUDE_USER_SETTING;
+
+public class SecuritySlowLogFieldProvider implements SlowLogFieldProvider {
+    private final Security plugin;
+    private boolean includeUserInIndexing = false;
+    private boolean includeUserInSearch = false;
+
+    public SecuritySlowLogFieldProvider(Security plugin) {
```

Comment:
`Security` is a `Plugin` subclass. Are you sure we want to depend on such a high level entry point class? When do we use this constructor?
We are 'providing' this instance via SPI, so the other noarg constructor is used , right?

Reply:
Thanks for calling this out. 

The one arg constructor is called from [here](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/plugins/PluginsService.java#L423),  where the parameter has to be a `Plugin` and the auth context that is needed is only available in the `Security` plugin. 

To pass "[SPI validation](https://github.com/elastic/elasticsearch/blob/main/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/JavaModulePrecommitTask.java#L141)" the provided implementation needs to be declared in `module-info` and one of the validation requirements is that it has a no arg constructor, but we never actually call it. I'll add an `IllegalStateException` to the no arg constructor like we do [here](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/ReservedSecurityStateHandlerProvider.java#L22) to make sure we don't end up in that state.